.github/workflows/build.yaml

```diff
@@ -0,0 +1,49 @@
+name: Build
+on:
+  pull_request:
+  push:
+    branches:
+      - confluence
+      - next
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          # Check https://github.com/livepeer/go-livepeer/pull/1891
+          # for ref value discussion
+          ref: ${{ github.event.pull_request.head.sha }}
+      - uses: actions/setup-node@v3
+        with:
+          node-version: "16"
+          cache: "yarn"
+      - name: Get yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+      - uses: actions/cache@v3
+        id: yarn-cache # use this to check for `cache-hit` (`steps.yarn-cache.outputs.cache-hit != 'true'`)
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+      - name: Install dependencies
+        run: |
+          yarn
+      - name: Linter
+        run: |
+          yarn lint
+      - name: Compile contracts
+        run: |
+          yarn compile
+      - name: 💡 Run contract tests with coverage reporting
+        run: |
+          yarn test:coverage
+      - name: Coveralls
```

Comment:
we can replace it with `codecov`, as we have better integration (along with more repositories) there.

Reply:
Created an issue here https://github.com/livepeer/protocol/issues/587